File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    static int dp[];
    private static int fibo(int n) {
        if(n == 0) return 0;
        if(n == 1) return 1;
        if(dp[n] == 0) {
            dp[n] = fibo(n - 1) % 1234567 + fibo(n - 2) % 1234567;
        }
        
        return dp[n] % 1234567;
    }
    public int solution(int n) {
        dp = new int[n + 1];
        return fibo(n);
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    var fibo = [];
    fibo[0] = 0;
    fibo[1] = 1;
    
    for(var i = 2;i <= n;i++) {
        fibo[i] = (fibo[i - 1] + fibo[i - 2]) % 1234567;
    }
    
    return fibo[n] % 1234567;
}